### PR TITLE
AP_HAL: Disco: move default param file path

### PR DESCRIPTION
**Pull request by mothacehe**
In future Disco releases, no ardupilot binary nor default param file
will be provided.

So move HAL_PARAM_DEFAULTS_PATH to ardupilot folder in RW memory.

### CODE REVIEW

**Comment by OXINARF** [comment]
Sad to know that ArduPilot won't be in Disco releases, but I like this change. I've always wanted to do it for Bebop but it wasn't liked at the time.

File: libraries/AP_HAL/board/linux.h [hunk 1]
```diff
@@ -111,7 +111,7 @@
 #define HAL_FLOW_PX4_BOTTOM_FLOW_VALUE_THRESHOLD 5000
 #define HAL_RCOUT_DISCO_BLDC_I2C_BUS 1
 #define HAL_RCOUT_DISCO_BLDC_I2C_ADDR 0x08
-#define HAL_PARAM_DEFAULTS_PATH "/etc/arduplane/disco.parm"
+#define HAL_PARAM_DEFAULTS_PATH "/data/ftp/internal_000/APM/disco.parm"
```
Comment (lucasdemarchi) [inline libraries/AP_HAL/board/linux.h]: can we replace APM there with ardupilot?

Comment (jberaud) [inline libraries/AP_HAL/board/linux.h]: Ok for me.

Comment (mothacehe) [inline libraries/AP_HAL/board/linux.h]: Ok then, I just posted two commits to do that.

**Comment by OXINARF** [comment]
Merged, thanks!

**Pull request merged**